python/cuml/naive_bayes/naive_bayes.py

```diff
@@ -0,0 +1,563 @@
+
+    from sklearn.datasets import fetch_20newsgroups
+    from sklearn.feature_extraction.text import CountVectorizer
+
+    from cuml.naive_bayes import MultinomialNB
+
+    # Load corpus
+
+    twenty_train = fetch_20newsgroups(subset='train',
+                              shuffle=True, random_state=42)
+
+    # Turn documents into term frequency vectors
+
+    count_vect = CountVectorizer()
+    features = count_vect.fit_transform(twenty_train.data)
+
+    # Put feature vectors and labels on the GPU
+
+    X = cp.sparse.csr_matrix(features.tocsr(), dtype=cp.float32)
+    y = cp.asarray(twenty_train.target, dtype=cp.int32)
+
+    # Train model
+
+    model = MultinomialNB()
+    model.fit(X, y)
+
+    # Compute accuracy on training set
+
+    model.score(X, y)
+
+    Output:
+
+    .. code-block:: python
+
+    0.9244298934936523
+
+    """
+    def __init__(self, alpha=1.0, fit_prior=True, class_prior=None):
+
+        """
+        Create new multinomial Naive Bayes instance
+
+        Parameters
+        ----------
+
+        alpha : float Additive (Laplace/Lidstone) smoothing parameter (0 for
+                no smoothing).
+        fit_prior : boolean Whether to learn class prior probabilities or no.
+                    If false, a uniform prior will be used.
+        class_prior : array-like, size (n_classes) Prior probabilities of the
+                      classes. If specified, the priors are not adjusted
+                      according to the data.
+        """
+
+        self.alpha = alpha
+        self.fit_prior = fit_prior
+        self.class_prior = class_prior
+
+        self.fit_called_ = False
+
+        self.classes_ = None
+        self.n_classes_ = 0
+
+        self.n_features_ = None
+
+    @cp.prof.TimeRangeDecorator(message="fit()", color_id=0)
+    def fit(self, X, y, sample_weight=None):
+        """
+        Fit Naive Bayes classifier according to X, y
+
+        Parameters
+        ----------
+
+        X : {array-like, cupy sparse matrix} of shape (n_samples, n_features)
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+        y : array-like shape (n_samples) Target values.
+        sample_weight : array-like of shape (n_samples)
+            Weights applied to individial samples (1. for unweighted).
+        """
+        return self.partial_fit(X, y, sample_weight)
+
+    @cp.prof.TimeRangeDecorator(message="fit()", color_id=0)
+    def _partial_fit(self, X, y, sample_weight=None, _classes=None):
+
+        if isinstance(X, np.ndarray):
+            X = rmm_cupy_ary(cp.asarray, X, X.dtype)
+        elif scipy.sparse.isspmatrix(X):
+            inds = rmm_cupy_ary(cp.asarray, X.indices, dtype=X.indices.dtype)
+            indptr = rmm_cupy_ary(cp.asarray, X.indptr, dtype=X.indptr.dtype)
+            data = rmm_cupy_ary(cp.asarray, X.data, dtype=X.data.dtype)
+            X = cp.sparse.csr_matrix((data, inds, indptr), shape=X.shape)
```

Comment:
This could be a future feature/follow up improvement, but similar to my comment above: Why _partial_fit only accepts host data?

Reply:
discussion offline: future work

Reply:
This was actually a really simple fix so I just went ahead and did it.